packages/app/components/Nav/Settings/Profile.tsx

```diff
@@ -0,0 +1,164 @@
+    console.log("permission not set > requesting");
+    await window.Notification.requestPermission();
+  }
+};
+
+export default function Settings() {
+  // first get initial data using useSWR (or from profile)
+  // todo: replace these constants with hits to useSWR call to profile
+  const user_id = 1;
+  const [desktopNotifToggled, setDesktopNotifToggled] = useState<boolean>(true);
+  const [phoneNotifToggled, setPhoneNotifToggled] = useState<boolean>(false);
+  const [phoneNumber, setPhoneNumber] = useState<string>("+12223334444");
+
+  const onDesktopNotifToggle = async (toggle: boolean) => {
+    if (toggle) {
+      check();
+      // try to get notification permissions
+      await requestNotificationPermission();
+      // get rid of old service worker, and then try and re-register.
+      // just kidding, this breaks Chrome for some reason (ai ya).
+      // unregister();
+      // have to use setTimeout because unregister does async things, but is sync
+      setTimeout(() => {
+        register();
+      }, 500);
+
+      setDesktopNotifToggled(true);
+    } else {
+      unregister();
+      setDesktopNotifToggled(false);
+    }
+  };
+
+  const onPhoneNotifToggle = async (toggle: boolean) => {
+    if (toggle) {
+      if (!phoneNumber) {
+        message.warning("Must enter phone number to enable text notifications");
+        setPhoneNotifToggled(false);
+      } else {
+        try {
+          //todo: change when ligma branch is merged
+          //await API.notif.phone.register(user_id, phoneNumbers[0]);
+          setPhoneNotifToggled(true);
+        } catch (e) {
+          message.error(e.body);
+          setPhoneNotifToggled(false);
+        }
+      }
+    } else {
+      //todo: add phone unregister endpoint
+      setPhoneNotifToggled(false);
+    }
+  };
+
+  return (
+    <div>
+      <Popover
+        content={
+          <PopoverContainer>
+            <Row>
+              <LableText> Web Notifications </LableText>
+              <SwitchContainer>
+                {" "}
+                <Switch
+                  checked={desktopNotifToggled}
+                  onChange={onDesktopNotifToggle}
+                />{" "}
+              </SwitchContainer>
+            </Row>
+            <Row>
+              <LableText> Text Notifications </LableText>
+              <SwitchContainer>
+                {" "}
+                <Switch
+                  checked={phoneNotifToggled}
+                  onChange={onPhoneNotifToggle}
+                />{" "}
+              </SwitchContainer>
+            </Row>
+            <Row>
+              <LableText> Phone </LableText>
+              <InputContainer>
+                {" "}
+                <Input
+                  value={phoneNumber}
+                  onChange={(newNum) => {
+                    setPhoneNumber(newNum.target.value);
+                  }}
+                  placeholder={"XXX-XXX-XXXX"}
+                />{" "}
```

Comment:
Do we need the {" "} in this file?

Reply:
I removed them and the UI looked the same to me, so I don't think so. removing them.